### PR TITLE
Fix redirection when we are in a directory

### PR DESCRIPTION
**Pull request by SebBareyre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Fix redirection when we are in a directory (module route for example)
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | -
| How to test?  | Try to access to a module route in directory structure that need to be logged in

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12920)

### CODE REVIEW

**Review by mickaelandrieu — DISMISSED** [review]

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -307,7 +307,7 @@ public function init()
         }
 
         if ($this->auth && !$this->context->customer->isLogged($this->guestAllowed)) {
-            Tools::redirect('index.php?controller=authentication' . ($this->authRedirection ? '&back=' . $this->authRedirection : ''));
+            Tools::redirect('/index.php?controller=authentication' . ($this->authRedirection ? '&back=' . $this->authRedirection : ''));
```
Comment (PierreRambaud) [inline classes/controller/FrontController.php]: What happened if your prestashop is under a subdirectory like `http://website.com/shop`

**Comment by PierreRambaud** [comment]
Need information before waiting for QA

**Review by Quetzacoalt91 — CHANGES_REQUESTED** [review]
Unfortunately, this change is in my opinion invalid.

I made a test from a ModuleFrontController which requires a customer to be logged in.
The URL generated on `develop` is:

```
http://[::1]/PrestaShop/fr/connexion
```

while with this PR we get
```
http://[::1]/PrestaShop/fr/index.php?controller==authentication
```

I unfortunately cannot reproduce the issue and this change brings another one. I'm not in favor in merging this PR. 


If you still have an issue on your side, could you consider this?

```php
            $request = [];
            if ($this->authRedirection) {
                $request['back'] = $this->authRedirection;
            }
            Tools::redirect($link->getPageLink('authentication', $useSSL, null, $request));
```

**Review by SebBareyre — COMMENTED** [review]
Here a fix of my PR

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
I still didn't manage to reproduce the issue, can you please create a reproducible issue?

**Comment by PierreRambaud** [comment]
Hi @SebBareyre ,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**